### PR TITLE
Closes #147 PointParametricEst implementation

### LINKED ISSUE #147: Designing PPE `VariableEstimate`
Parametric Point Estimates (PPE) are stored in `VariableEstimate`, but there is a question of what data should be stored with it?  

Keeping in mind the data is stored in dictionary, example from current design:
```julia
 var.estimateDict[:default] = Dict{Symbol, VariableEstimate}(
    :max => VariableEstimate(:default, :max, varMax),
    :mean => VariableEstimate(:default, :mean, varMean),
    :modefit => VariableEstimate(:default, :modefit, varModefit))
```

Suggestions are:
- [x] pppeType, `:mean / :max / :modefit`
- [x] actual value 
- [ ] softtypes which can point to (`:Pose2`) or contain manifolds (`Pose2().manifolds`)
- [x] `solveKey`
- [x] `lastupdated::DateTime`
...

Current here:
https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/blob/bca01e6188ef206122a7c9a51378ca3d0f1e0b8c/src/entities/DFGVariable.jl#L76-L82

### PR DESCRIPTION
**Pull request by Affie**
Closes #147 
Closes #158 replace pr with new design
Closes #148 replace pr with new design

**Note**
Just the struct so far, TODO integrate, testing, compare, etc.

### CODE REVIEW

File: src/entities/DFGVariable.jl [hunk 1]
```diff
@@ -93,7 +95,7 @@ mutable struct DFGVariable <: AbstractDFGVariable
     label::Symbol
     timestamp::DateTime
     tags::Vector{Symbol}
-    estimateDict::Dict{Symbol, Dict{Symbol, <: AbstractVariableEstimate}}
+    estimateDict::Dict{Symbol, <: AbstractVariableEstimate}
```
Comment (dehann) [inline src/entities/DFGVariable.jl]: have we figured out if we going with `<: AbstractParametricPointEst` yet?

see: https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/issues/147#issuecomment-545639185

Comment (dehann) [reply src/entities/DFGVariable.jl]: you are using it above, so might just be a mistake or am i missing something?

Comment (Affie) [reply src/entities/DFGVariable.jl]: Just missed it thanks. will fix

File: src/services/DFGVariable.jl [hunk 2]
```diff
@@ -112,15 +113,11 @@ function ==(a::VariableNodeData,b::VariableNodeData, nt::Symbol=:var)
 end
 
 """
-    $(SIGNATURES)
-Equality check for VariableEstimate.
+    ==(x::T, y::T) where T <: AbstractPointParametricEst
```
Comment (dehann) [inline src/services/DFGVariable.jl]: missed this yesterday sorry, just want to make sure the `import Base: ==` is there, not seeing in the PR.

Comment (dehann) [reply src/services/DFGVariable.jl]: this is the closest I can find:
https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/blob/9afd7a380cc5d124c74bc3a1a466202cd5370fe2/src/LightDFG/FactorGraphs/FactorGraphs.jl#L5

Comment (GearsAD) [reply src/services/DFGVariable.jl]: Yep, top of the file:
```
import Base: ==, convert
```

**Comment by GearsAD** [comment]
Very cool!

**Pull request merged**